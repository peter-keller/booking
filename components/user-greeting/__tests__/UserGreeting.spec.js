import { shallowMount } from '@vue/test-utils'

import UserGreeting from '..'

describe('Component: UserGreeting', () => {
  const factory = (opts = {}) => shallowMount(UserGreeting, {
    ...opts
  })

  it('matches snapshot', () => {
    const wrapper = factory()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('calls "OpenSidebar" when Icon clicked', () => {
    const wrapper = factory()

    wrapper.find('icon-stub').vm.$emit('click')

    expect(wrapper.emitted('click')).toBeTruthy()
  })
})
