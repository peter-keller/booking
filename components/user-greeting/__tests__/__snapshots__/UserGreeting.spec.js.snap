// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserGreeting matches snapshot 1`] = `
"<div class=\\"user-greeting\\">
  <div class=\\"user-greeting__message\\">
    <h3>HOMEPAGE_WELCOME_MESSAGE</h3>
    <p>Patryk</p>
  </div>
  <icon-stub name=\\"ArrowDown\\" size=\\"medium\\" stroke=\\"\\" clickable=\\"true\\"></icon-stub>
</div>"
`;
